db/compaction_job.cc

```diff
@@ -1144,17 +1146,24 @@ Status CompactionJob::FinishCompactionOutputFile(
                      meta->marked_for_compaction ? " (need compaction)" : "");
     }
   }
-  std::string fname = TableFileName(db_options_.db_paths, meta->fd.GetNumber(),
-                                    meta->fd.GetPathId());
+  std::string fname;
+  FileDescriptor output_fd;
+  if (meta != nullptr) {
```

Comment:
can you just add `assert(meta != nullptr)` to fix this? it's a pointer to a stack object so should be ok

Reply:
Why does it a stack object? It is not clear to me when I look at the code. I guess it is not clear to CLANG either.